### PR TITLE
plat/drivers/virtio-blk: Remove uk_free call on request dequeue

### PR DESCRIPTION
**Pull request by jschlumpp**
### Prerequisite checklist

 - [x] Read the [contribution guidelines](https://docs.unikraft.org/contribute.html) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [x] Ran the [`checkpatch.pl`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.pl) on your commit series before opening this PR;
 - [ ] Updated relevant documentation.

### Base target

 - Architecture(s): N/A
 - Platform(s): kvm
 - Application(s): N/A

### Additional configuration

 - `CONFIG_VIRTIO_BLK=y`

### Description of changes

The virtio_blkdev_queue_dequeue function can be called in an interrupt
handler context and calling free there can corrupt allocators such as
ukallocbbuddy. This creates an additional list for delaying the free
call.

This PR supersedes #333.

### CODE REVIEW

**Review by craciunoiuc — APPROVED** [review]
Heyo @mschlumpp @marcrittinghaus,

All good here.

Reviewed-by: Cezar Craciunoiu <cezar.craciunoiu@gmail.com>

File: plat/drivers/virtio/virtio_blk.c [hunk 1]
```diff
@@ -102,10 +102,16 @@ struct uk_blkdev_queue {
 	/* The scatter list and its associated fragments */
 	struct uk_sglist sg;
 	struct uk_sglist_seg *sgsegs;
+	/* List of virtio_blkdev_requests to be freed outside of interrupt
+	 * handler
+	 */
+	/* TODO: Replace with Linux llist-style list for SMP support */
```
Comment (nderjung) [inline plat/drivers/virtio/virtio_blk.c:108]:
```diff
- /* TODO: Replace with Linux llist-style list for SMP support */
+ /* TODO: Replace with Linux list-style list for SMP support */
```

**Review by marcrittinghaus — APPROVED** [review]
@mschlumpp Thank you very much! This fixes an important problem 😃

Approved-by: Marc Rittinghaus <marc.rittinghaus@kit.edu>

**Review by nderjung — DISMISSED** [review]
Thanks! This works my side, just one spelling mistake, see below:

**Comment by jschlumpp** [comment]
> Thanks! This works my side, just one spelling mistake, see below:

It is actually not only linked-list but [lock-less linked-list](https://github.com/torvalds/linux/blob/master/include/linux/llist.h)

**Comment by marcrittinghaus** [comment]
> > Thanks! This works my side, just one spelling mistake, see below:
> 
> It is actually not only linked-list but [lock-less linked-list](https://github.com/torvalds/linux/blob/master/include/linux/llist.h)

Perhaps you should have named it `llllist`. Then everyone would have known what you mean 😆

**Pull request closed without merging**